### DISCUSSION TITLE
How to switch frames with a press of a button

### DISCUSSION THREAD

**Discussion by SamarthaJ** [TomSchimansky/CustomTkinter#2262] [Q&A] [answered]
Hello there, I am new to this and I'm using this lib for to create gui for one of my project, I have a 3 frames navbar, sidebar, mainboard. I need to know is there a way to switch/ replace mainboard with someother CTKFrame with a click of a button???

**Comment by JanPanthera** [comment, marked as answer]
I use below code for my tabview

```
    def show_tab(self, frame):
        for frm in self.btn_to_frame.values():
            frm.grid_remove()
        frame.grid(row=1, column=0, columnspan=3, sticky="nsew")

    def hide_tab(self, frame):
        frame.grid_remove()
```
If you use a grid system then you can simply call grid_remove(), this hides the frame by removing it from the screen, basically hiding it.
To show it again, just add it to the master/frame again

**Reply by SamarthaJ** [reply]
Thanks for this, but I have defined the each frames in different classes and I cannot access it can you help with this

Thanks in advance😊

**Reply by JanPanthera** [reply]
if you can provide the code, i can take a look
i also have frames in its own class, i have my ui code inside a json file. ;-)

**Reply by SamarthaJ** [reply]
main.py is in ui

[Health Management system.zip](https://github.com/TomSchimansky/CustomTkinter/files/14322606/Health.Management.system.zip)

**Reply by Anonymous6598** [reply]
Great job @JanPanthera!

**Comment by JanPanthera** [comment]
[Health Management system.zip](https://github.com/TomSchimansky/CustomTkinter/files/14322953/Health.Management.system.zip)

**Reply by JanPanthera** [reply]
this should be the modified one xD

**Reply by SamarthaJ** [reply]
Thanks dude XD I'm new to GUI in Python so I didn't know what to do BTW Thanks buddy you saved my project scores 💀

**Reply by JanPanthera** [reply]
no problem, if you have any other questions abou ui, just ask, i love to help where i can :-)
